specifyweb/frontend/js_src/lib/components/DataModel/specifyTable.ts

```diff
@@ -241,9 +241,10 @@ export class SpecifyTable<SCHEMA extends AnySchema = AnySchema> {
     });
 
     const useLabels = getCache('forms', 'useFieldLabels') ?? true;
-    this.localization = getSchemaLocalization()[this.name.toLowerCase()] ?? {
-      items: {},
-    };
+    this.localization = getSchemaLocalization()[this.name.toLowerCase()] ??
+      getSchemaLocalization()[this.name] ?? {
+        items: {},
+      };
```

Comment:
If we do need a workaround on the frontend, would `caseInsensitiveHash` be fine here?
(Though I am mostly worried about the difference in performance with `caseInsensitiveHash` in this case, so the way it is currently is probably fine IF a workaround is needed). 

```suggestion
    this.localization = caseInsensitiveHash(
      getSchemaLocalization(),
      this.name
    ) ?? {
      items: {},
    };
```